### PR TITLE
Include gocritic, gosec and staticcheck versions

### PR DESCRIPTION
**Pull request by jasonodoom**
#  #

## 🗣 Description ##

This PR is to track the versions for [GoCritic](https://github.com/golang/mock), [GoSec](https://github.com/securego/gosec) and [Staticcheck](https://github.com/dominikh/go-tools) into this repo so it can be consumed/used in GitHub Actions workflows. 

## 💭 Motivation and context ##

This change is based on current team practices. 

## 🧪 Testing ##

Pre-commit verified syntactical accuracy. 

## ✅ Pre-approval checklist ##

- [X] This PR has an informative and human-readable title.
- [X] Changes are limited to a single goal - *eschew scope creep!*
- [X] All relevant type-of-change labels have been added.
- [X] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [X] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [X] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,9 +13,12 @@ None
 
 | Name              | Description |
 |-------------------|-------------|
+| go-critic-version | The version of [go-critic](https://github.com/go-critic/go-critic) to use. |
 | go-junit-report-version | The version of [go-junit-report](https://github.com/jstemmer/go-junit-report) to use. |
 | gomock-version | The version of [GoMock](https://github.com/golang/mock) to use. |
+| gosec-version | The version of [GoSec](https://github.com/securego/gosec) to use. |
 | packer-version | The version of [Packer](https://packer.io) to use. |
+| go-staticcheck-version | The version of [Staticcheck](https://github.com/dominikh/go-tools) to use. |
 | shfmt-version | The version of [shfmt](https://github.com/mvdan/sh#shfmt) to use. |
```
Comment (mcdonnnj) [inline README.md]: Adjusting the output name to match the tool, sorting alphabetically, and adjusting the URL to be tool specific.

```suggestion
| shfmt-version | The version of [shfmt](https://github.com/mvdan/sh#shfmt) to use. |
| staticcheck-version | The version of [staticcheck](https://github.com/dominikh/go-tools/tree/master/cmd/staticcheck) to use. |
```

File: action.yml [hunk 2]
```diff
@@ -6,18 +6,27 @@ branding:
...
+    value: ${{ steps.gosec.outputs.version }}
   packer-version:
     description: The version of Packer to use.
     value: ${{ steps.packer.outputs.version }}
   shfmt-version:
     description: The version of shfmt to use.
     value: ${{ steps.shfmt.outputs.version }}
+  go-staticcheck-version:
+    description: The version of dominikh/go-tools to use.
+    value: ${{ steps.go-staticcheck.outputs.version }}
```
Comment (mcdonnnj) [inline action.yml]:
```suggestion
  staticcheck-version:
    description: The version of staticcheck to use.
    value: ${{ steps.staticcheck.outputs.version }}
```

File: action.yml [hunk 3]
```diff
@@ -28,18 +37,27 @@ outputs:
...
+    - id: gosec
+      run: echo "version=v2.16.0" >> $GITHUB_OUTPUT
+      shell: bash
     - id: packer
       run: echo "version=1.8.2" >> $GITHUB_OUTPUT
       shell: bash
     - id: shfmt
       run: echo "version=v3.5.1" >> $GITHUB_OUTPUT
       shell: bash
+    - id: go-staticcheck
```
Comment (mcdonnnj) [inline action.yml]:
```suggestion
    - id: staticcheck
```

File: action.yml [hunk 4]
```diff
@@ -28,18 +37,27 @@ outputs:
...
+      run: echo "version=v1.3.0" >> $GITHUB_OUTPUT
```
Comment (mcdonnnj) [inline action.yml]: When looking at these tools this was the latest tag available. Is there something I missed?

```suggestion
      run: echo "version=v0.4.3" >> $GITHUB_OUTPUT
```

File: action.yml [hunk 5]
```diff
@@ -6,18 +6,27 @@ branding:
   color: purple
 description: Setup a shared GitHub Actions workflow environment.
 outputs:
+  go-critic-version:
+    description: The version of go-critic/go-critic to use.
```
Comment (mcdonnnj) [inline action.yml]:
```suggestion
    description: The version of go-critic to use.
```

File: action.yml [hunk 6]
```diff
@@ -28,18 +37,27 @@ outputs:
 runs:
   using: "composite"
   steps:
+    - id: go-critic
+      run: echo "version=0.8.1" >> $GITHUB_OUTPUT
```
Comment (mcdonnnj) [inline action.yml]:
```suggestion
      run: echo "version=v0.8.1" >> $GITHUB_OUTPUT
```

**Review by mcdonnnj — CHANGES_REQUESTED** [review]

**Comment by jsf9k** [comment]
@jasonodoom - Please note that I added back into the PR description a checkbox that you deleted in what I assume was an accident of haste.

File: action.yml (hunk 4 above)
Comment (jasonodoom) [reply action.yml]: No, it's something I missed. Thanks!

**Comment by jasonodoom** [comment]
> @jasonodoom - Please note that I added back into the PR description a checkbox that you deleted in what I assume was an accident of haste.

@jsf9k It definitely was. 😓 Thank you!

**Review by jsf9k — CHANGES_REQUESTED** [review]
I don't think this PR should be merged until [this conversation](https://github.com/cisagov/skeleton-generic/pull/138#discussion_r1259813532) is resolved.

**Review by jsf9k — APPROVED** [review]

**Comment by jsf9k** [comment]
> I don't think this PR should be merged until [this conversation](https://github.com/cisagov/skeleton-generic/pull/138#discussion_r1259813532) is resolved.

That conversation is now resolved.

**Review by dav3r — APPROVED** [review]
LGTM! :goberserk:

**Comment by jasonodoom** [comment]
> I don't think this PR should be merged until [this conversation](https://github.com/cisagov/skeleton-generic/pull/138#discussion_r1259813532) is resolved.

@jsf9k but  doesn't this have to be merged first to pull in the defined package versions?  I can actually test in a different branch but the builds from generic will fail if this isn't merged first.

**Comment by jsf9k** [comment]
> > I don't think this PR should be merged until [this conversation](https://github.com/cisagov/skeleton-generic/pull/138#discussion_r1259813532) is resolved.
> 
> @jsf9k but doesn't this have to be merged first to pull in the defined package versions? I can actually test in a different branch but the builds from generic will fail if this isn't merged first.

@jasonodoom - Please see [here](https://github.com/cisagov/skeleton-generic/pull/138#issuecomment-1632509355).  I temporarily modified the GitHub Actions workflow in cisagov/skeleton-generic to use this branch to get around this very issue.

File: action.yml [hunk 7]
```diff
@@ -28,18 +37,27 @@ outputs:
...
+      run: echo "version=v2.16.0" >> $GITHUB_OUTPUT
+      shell: bash
     - id: packer
       run: echo "version=1.8.2" >> $GITHUB_OUTPUT
       shell: bash
     - id: shfmt
       run: echo "version=v3.5.1" >> $GITHUB_OUTPUT
       shell: bash
+    - id: staticcheck
+      run: echo "version=2023.1.3" >> $GITHUB_OUTPUT
```
Comment (mcdonnnj) [inline action.yml]: Why was this version changed? The old version appears to have worked fine in https://github.com/cisagov/skeleton-generic/pull/138 per the [run log](https://github.com/cisagov/skeleton-generic/actions/runs/5536385486/jobs/10103987744#step:14:41).

Comment (jsf9k) [inline action.yml]: Whichever version of staticcheck you end up on, there is some commit history to be cleaned up here.  (Commits 265128b, 9fe08e9, and c2ebd61.)

**Review by jsf9k — CHANGES_REQUESTED** [review]

File: action.yml (hunk 7 above)
Comment (jsf9k) [reply action.yml]: I think you can just remove commits 9fe08e9 and 73eb498, no?

Comment (jasonodoom) [reply action.yml]: There was some confusion in regards to which version to reference but I went ahead and used the one shown in the documentation as opposed to the one linked. Previously, `v0.4.3` was failing so I changed the version number. Maybe it was unrelated.

Comment (mcdonnnj) [reply action.yml]: I believe it was unrelated and I would prefer the semver tag. That provides clearer information about what kinds of changes to expect when updating versions.

Comment (jasonodoom) [reply action.yml]: @jsf9k do you mean combine them into one to remove duplicate commits?

Comment (jsf9k) [reply action.yml]: - Commit 265128b changes the version from `v0.4.3` to `2023.1.3`
- Commit 9fe08e9 changes the version from `2023.1.3` to `v0.4.3`
- Commit 73eb498 changes the version from `v0.4.3` to `2023.1.3`

So you could rebase and drop commits 9fe08e9 and 73eb498 with no change to the actual code.

Comment (jsf9k) [reply action.yml]: Although I see that [@mcdonnnj prefers the semver version](https://github.com/cisagov/setup-env-github-action/pull/65#discussion_r1262947618) (as do I), so if you follow that advice you could rebase and drop _all three_ commits.

Comment (jsf9k) [reply action.yml]: I also prefer the semver tag.

Comment (jasonodoom) [reply action.yml]: @jsf9k  Please see https://github.com/cisagov/setup-env-github-action/pull/65/commits/c625d814f0afdbed58563decc5ad19bc2fdabba0

Comment (jasonodoom) [reply action.yml]: Please see https://github.com/cisagov/setup-env-github-action/commit/c625d814f0afdbed58563decc5ad19bc2fdabba0

**Review by jsf9k — APPROVED** [review]

File: README.md [hunk 8]
```diff
@@ -13,10 +13,13 @@ None
 
 | Name              | Description |
 |-------------------|-------------|
+| go-critic-version | The version of [go-critic](https://github.com/go-critic/go-critic) to use. |
 | go-junit-report-version | The version of [go-junit-report](https://github.com/jstemmer/go-junit-report) to use. |
 | gomock-version | The version of [GoMock](https://github.com/golang/mock) to use. |
+| gosec-version | The version of [GoSec](https://github.com/securego/gosec) to use. |
```
Comment (mcdonnnj) [inline README.md]: This aligns both with the project's capitalization and the description for the Action's output.

```suggestion
| gosec-version | The version of [gosec](https://github.com/securego/gosec) to use. |
```

**Review by mcdonnnj — APPROVED** [review]
Just one adjustment but otherwise LGTM ✔

**Review by mcdonnnj — APPROVED** [review]
LGTM ✔

**Pull request merged**